File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        stack = []
        for char in s:
            if char != ")":
                stack.append(char)
                continue
            substr = []
            while stack[-1] != "(":
                substr.append(stack.pop())
            stack.pop()
            stack.extend(substr)
        return "".join(stack)

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        string result;
        stack<int> opens;
        for (const char c : s) {
            if (c == '(') {
                opens.push(result.size());
            } else if (c == ')') {
                reverse(result.begin() + opens.top(), result.end());
                opens.pop();
            } else {
                result.push_back(c);
            }
        }
        return result;
    }
};
